hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/HoodieAsyncService.java

```diff
@@ -141,35 +140,17 @@ public void start(Function<Boolean, Boolean> onShutdownCallback) {
   protected abstract Pair<CompletableFuture, ExecutorService> startService();
 
   /**
-   * A monitor thread is started which would trigger a callback if the service is shutdown.
+   * Add shutdown callback for the completable future.
    * 
-   * @param onShutdownCallback
+   * @param callback The callback
    */
-  private void monitorThreads(Function<Boolean, Boolean> onShutdownCallback) {
-    LOG.info("Submitting monitor thread !!");
-    Executors.newSingleThreadExecutor(r -> {
-      Thread t = new Thread(r, "Monitor Thread");
-      t.setDaemon(isRunInDaemonMode());
```

Comment:
Here the code new a thread pool but never shut down it, the core working thread always lives there thus leak ~

Reply:
yes this seems okay